cloudwatch_logs_common/src/log_publisher.cpp

```diff
@@ -108,102 +107,137 @@ Aws::CloudWatchLogs::ROSCloudWatchLogsErrors LogPublisher::StopPublisherThread()
+  if (CW_LOGS_SUCCEEDED == check_log_stream_exists_status) {
+    this->run_state_ = Aws::CloudWatchLogs::LOG_PUBLISHER_RUN_INIT_TOKEN;
+    AWS_LOGSTREAM_DEBUG(__func__, "Found existing log stream: " << log_stream_);
+    return;
+  }
+  Aws::CloudWatchLogs::ROSCloudWatchLogsErrors create_log_stream_status =
+    this->cloudwatch_facade_->CreateLogStream(this->log_group_, this->log_stream_);
+  if (CW_LOGS_SUCCEEDED == create_log_stream_status) {
+    this->run_state_ = Aws::CloudWatchLogs::LOG_PUBLISHER_RUN_INIT_TOKEN;
+    AWS_LOG_DEBUG(__func__, "Successfully created log stream.");
+  } else if (CW_LOGS_LOG_STREAM_ALREADY_EXISTS == create_log_stream_status) {
+    this->run_state_ = Aws::CloudWatchLogs::LOG_PUBLISHER_RUN_INIT_TOKEN;
+    AWS_LOG_DEBUG(__func__, "Log stream already exists, quit attempting to create it.");
+  } else {
+    AWS_LOGSTREAM_ERROR(__func__, "Failed to create log stream with , retrying ...");
+  }
+}
 
-    this->cloudwatch_facade_->GetLogStreamToken(this->log_group_, this->log_stream_, next_token);
-
-    auto shared_logs_obj = this->shared_logs_.load(std::memory_order_acquire);
-    if (nullptr != shared_logs_obj) {
-      std::list<Aws::CloudWatchLogs::Model::InputLogEvent> * logs;
-      publisher_status = shared_logs_obj->getDataAndLock(logs);
-      if (CW_LOGS_SUCCEEDED == publisher_status) {
-        if (!logs->empty()) {
-          Aws::CloudWatchLogs::ROSCloudWatchLogsErrors send_logs_status = CW_LOGS_FAILED;
-          int tries = kMaxRetries;
-
-          while (CW_LOGS_SUCCEEDED != send_logs_status && tries > 0) {
-            send_logs_status = this->cloudwatch_facade_->SendLogsToCloudWatch(
-              next_token, this->log_group_, this->log_stream_, logs);
-            if (CW_LOGS_SUCCEEDED != send_logs_status) {
-              AWS_LOG_WARN(__func__, "Unable to send logs to CloudWatch, retrying ...");
-              Aws::CloudWatchLogs::ROSCloudWatchLogsErrors get_token_status =
-                this->cloudwatch_facade_->GetLogStreamToken(this->log_group_, this->log_stream_,
-                                                            next_token);
-              if (CW_LOGS_SUCCEEDED != get_token_status) {
-                AWS_LOG_WARN(__func__,
-                             "Unable to obtain the sequence token to use, quit attempting to send "
-                             "logs to CloudWatch");
-                break;
-              }
-            }
-            tries--;
-          }
+void LogPublisher::InitToken(Aws::String & next_token)
+{
+  Aws::CloudWatchLogs::ROSCloudWatchLogsErrors get_token_status =
+    this->cloudwatch_facade_->GetLogStreamToken(this->log_group_, this->log_stream_,
+                                                next_token);
+  if (CW_LOGS_SUCCEEDED == get_token_status) {
+    this->run_state_ = Aws::CloudWatchLogs::LOG_PUBLISHER_RUN_SEND_LOGS;
+  } else {
+    AWS_LOG_WARN(__func__, "Unable to obtain the sequence token to use");
```

Comment:
Only a warning? Can logging continue without it? Could you also log out the error received here.

Reply:
Updated to error.